### ISSUE TITLE
Won't run with rbenv

### ISSUE THREAD

**Issue by mconigliaro** [badmadrad/SublimeLinter-foodcritic#2]
```
SublimeLinter: foodcritic version query: /usr/local/opt/rbenv/shims/ruby /usr/local/opt/rbenv/shims/foodcritic --version 
SublimeLinter: WARNING: no foodcritic version could be extracted from:
ruby: no Ruby script found in input (LoadError)
```

Here's what happens when you do that on the command line:

```
$ /usr/local/opt/rbenv/shims/ruby /usr/local/opt/rbenv/shims/foodcritic --version
ruby: no Ruby script found in input (LoadError)
```

Why not just run the foodcritic binary without specifying the interpreter? i.e.

```
 /usr/local/opt/rbenv/shims/foodcritic --version
```

**Comment by mar-io** [comment]
I think this is more of an issue with sublime linter/rbenv compatibility. If you are not completely committed to rbenv I would recommend checking out chruby/ruby-install which is awesome and works for me. Can you read through this and see if it helps?

https://sunlightfoundation.com/blog/2015/03/31/making-sublimelinter-work-with-rbenv-and-rubocop/

I think you may need to add rbenv path to sublime linter or set/install a global ruby and install the gem to that specific ruby version.

**Comment by mconigliaro** [comment]
Thanks for the link, but I'm actually already doing all that stuff (my other plugins wouldn't work otherwise). My rbenv is loaded via `.zprofile` so ruby and all the global gems (including rubocop and foodcritic) are available to Sublime. The problem I see in the debug console is that the version query for your foodcritic plugin:

```
SublimeLinter: foodcritic version query: /usr/local/opt/rbenv/shims/ruby /usr/local/opt/rbenv/shims/foodcritic --version
```

is different than rubocop's for some reason:

```
SublimeLinter: rubocop version query: /usr/local/opt/rbenv/shims/ruby -S rubocop --version 
```

Looks like if your plugin did the same thing as rubocop, it would work for me too:

```
/usr/local/opt/rbenv/shims/ruby -S foodcritic --version
foodcritic 6.1.1
```

Unfortunately, it's not so easy for me to switch to chruby at the moment.

**Comment by mconigliaro** [comment]
Looks like the rubocop plugin had a similar bug last year. Here's their fix: https://github.com/SublimeLinter/SublimeLinter-rubocop/pull/18

**Comment by mar-io** [comment]
Yeah it does appear rubocop executes the gem differently. I will work on a fix.

**Comment by mar-io** [comment]
I just pushed a new version with change I think will fix the rbenv issue. Please reinstall or wait until the package is updated and try again and let me know.

**Comment by mconigliaro** [comment]
Awesome, thanks for the help on this! Unfortunately, now I'm getting a slightly different error:

```
SublimeLinter: foodcritic version query: /usr/local/opt/rbenv/shims/ruby /usr/local/opt/rbenv/shims/foodcritic -S foodcritic --version 
SublimeLinter: WARNING: no foodcritic version could be extracted from:
ruby: no Ruby script found in input (LoadError)
```

Seems like the `foodcritic` command is being expanded to its full path somehow.

Based on what I see in [SublimeLinter-rubocop](https://github.com/SublimeLinter/SublimeLinter-rubocop/blob/master/linter.py) and [the docs](http://www.sublimelinter.com/en/latest/linter_attributes.html), I think maybe `cmd` should be set to `['ruby', '-S', 'foodcritic', ... ]`? If I get a few minutes, I'll see if I can figure out how to test this myself and submit a PR. Interesting that the docs also explicitly [claim to support rvm and rbenv without you having to do anything special](http://www.sublimelinter.com/en/latest/ruby_linter.html#rbenv-and-rvm-support).

**Comment by mar-io** [comment]
No need for a pull request on that. I also installed rbenv on vm and noticed the following in my console when I added the "ruby -S":

```
    SublimeLinter: rubocop activated: ['/home/badbox2/.rbenv/shims/ruby'] 
    SublimeLinter: foodcritic activated: ['/home/badbox2/.rbenv/shims/ruby'
```

Notice the /shims/ruby/ which is same for both rubocop and foodcritic. So I just tagged  that change and the release should get picked up by package control soon. I would be surprised if this doesn't load it for you. If this doesn't work I would probably need a pull request.
